### PR TITLE
amadeus: Improve and fix delay effect processing

### PR DESCRIPTION
**Pull request by Thog**
This rework the delay effect processing by representing calculation with the appropriate matrix and by unrolling some loops in the code.
This allows better optimization by the JIT while making it more readable.

Also fix a bug in the Surround code path found while looking back at my notes.

### CODE REVIEW

File: Ryujinx.Audio/Renderer/Dsp/Command/DelayCommand.cs [hunk 1]
```diff
@@ -78,133 +80,152 @@ private unsafe void ProcessDelayMono(ref DelayState state, float* outputBuffer,
...
-            Span<float> temp = stackalloc float[channelCount];
-
             float delayFeedbackBaseGain = state.DelayFeedbackBaseGain;
             float delayFeedbackCrossGain = state.DelayFeedbackCrossGain;
             float inGain = FixedPointHelper.ToFloat(Parameter.InGain, FixedPointPrecision);
             float dryGain = FixedPointHelper.ToFloat(Parameter.DryGain, FixedPointPrecision);
             float outGain = FixedPointHelper.ToFloat(Parameter.OutGain, FixedPointPrecision);
 
+            Matrix4x4 delayFeedback = new Matrix4x4(
+                                                        delayFeedbackBaseGain , delayFeedbackCrossGain, delayFeedbackCrossGain, 0.0f,
```
Comment (gdkchan) [inline Ryujinx.Audio/Renderer/Dsp/Command/DelayCommand.cs:140]: I'd align line this:
```cs
Matrix4x4(delayFeedbackBaseGain , delayFeedbackCrossGain, delayFeedbackCrossGain, 0.0f,
```
And on last line:
```cs
0.0f                  , delayFeedbackCrossGain, delayFeedbackCrossGain, delayFeedbackBaseGain);
```

File: Ryujinx.Audio/Renderer/Utils/Math/Matrix2x2.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+            m.M11 = value1.M11 * value2;
+            m.M12 = value1.M12 * value2;
+            m.M21 = value1.M21 * value2;
+            m.M22 = value1.M22 * value2;
+
+            return m;
+        }
+
+        public static unsafe Matrix2x2 operator *(Matrix2x2 value1, Matrix2x2 value2)
```
Comment (gdkchan) [inline Ryujinx.Audio/Renderer/Utils/Math/Matrix2x2.cs:58]: Why is that unsafe? The code doesn't look like it needs unsafe.

File: Ryujinx.Audio/Renderer/Utils/Math/Matrix2x2.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
+﻿using System;
+
+namespace Ryujinx.Audio.Renderer.Utils.Math
+{
+    public record struct Matrix2x2
```
Comment (gdkchan) [inline Ryujinx.Audio/Renderer/Utils/Math/Matrix2x2.cs:5]: Type should not be public.

File: Ryujinx.Audio/Renderer/Utils/Math/Matrix6x6.cs [hunk 4]
```diff
@@ -0,0 +1,97 @@
+﻿namespace Ryujinx.Audio.Renderer.Utils.Math
+{
+    public record struct Matrix6x6
```
Comment (gdkchan) [inline Ryujinx.Audio/Renderer/Utils/Math/Matrix6x6.cs:3]: Type should not be public.

File: Ryujinx.Audio/Renderer/Utils/Math/Vector6.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
+﻿using System.Runtime.CompilerServices;
+
+namespace Ryujinx.Audio.Renderer.Utils.Math
+{
+    public struct Vector6
```
Comment (gdkchan) [inline Ryujinx.Audio/Renderer/Utils/Math/Vector6.cs:5]: Type should not be public.

File: Ryujinx.Audio/Renderer/Utils/Math/Vector6.cs [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    {
+        public float X;
+        public float Y;
+        public float Z;
+        public float W;
+        public float V;
```
Comment (gdkchan) [inline Ryujinx.Audio/Renderer/Utils/Math/Vector6.cs:11]: Is that the "standard" naming for a 6 elements vector?

File: Ryujinx.Audio/Renderer/Utils/Math/Vector6.cs [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+                left.W + right.W,
+                left.V + right.V,
+                left.U + right.U
+            );
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector6 operator *(Vector6 left, Vector6 right)
+        {
+            return new Vector6(
```
Comment (gdkchan) [inline Ryujinx.Audio/Renderer/Utils/Math/Vector6.cs:44]: Same comment I made about matrix regarding alignment applies here too.

File: Ryujinx.Audio/Renderer/Utils/Math/Vector6.cs (hunk 6 above)
Comment (Thog) [reply Ryujinx.Audio/Renderer/Utils/Math/Vector6.cs:11]: Not really entirely sure if there is a standard way here tbh

Comment (merryhime) [reply Ryujinx.Audio/Renderer/Utils/Math/Vector6.cs:11]: For a 6-vector I've most often seen (x₁ x₂ x₃ x₄ x₅ x₆).

**Review by AcK77 — APPROVED** [review]

**Review by gdkchan — APPROVED** [review]
lgtm, thanks.

**Pull request merged**